### PR TITLE
Add setting showPythonInlineValues

### PR DESCRIPTION
**Pull request by paulacamargo25**
Add setting for showing the inline values.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -131,6 +131,12 @@
                     "description": "%debugpy.debugJustMyCode%",
                     "scope": "resource",
                     "type": "boolean"
+                },
+                "debugpy.showPythonInlineValues": {
+                    "default": false,
+                    "description": "%debugpy.showPythonInlineValues%",
+                    "scope": "resource",
+                    "type": "boolean"
```
Comment (karthiknadig) [inline package.json:139]: This needs a `experimental ` tag.

**Review by karthiknadig — CHANGES_REQUESTED** [review]

**Review by connor4312 — COMMENTED** [review]
Is there a request to disable/enable debug inline values on a per-language basis versus the base `debug.inlineValues`? If so this is something we should probably have in VS Code itself rather than an extension-specific setting

**Comment by paulacamargo25** [comment]
> Is there a request to disable/enable debug inline values on a per-language basis versus the base `debug.inlineValues`? If so this is something we should probably have in VS Code itself rather than an extension-specific setting

We are using this setting as experimental, there are some variables in the python inline values that we need to review and investigate more for the next iteration. So we decided to add this setting to have this value set to `false` by default.

**Comment by connor4312** [comment]
It looks like `debug.inlineValues` functions in the current stable version of the Python extension, would a default value of `false` disable them for everyone?

**Comment by paulacamargo25** [comment]
> It looks like `debug.inlineValues` functions in the current stable version of the Python extension, would a default value of `false` disable them for everyone?

Yes, it will disable all the inline values

File: package.nls.json [hunk 2]
```diff
@@ -4,5 +4,6 @@
     "debugpy.command.debugUsingLaunchConfig.title": "Python Debugger: Debug using launch.json",
     "debugpy.command.reportIssue.title": "Report Issue...",
     "debugpy.command.viewOutput.title": "Show Output",
-    "debugpy.debugJustMyCode": "When debugging only step through user-written code. Disable this to allow stepping into library code."
+    "debugpy.debugJustMyCode": "When debugging only step through user-written code. Disable this to allow stepping into library code.",
+    "debugpy.showPythonInlineValues": "Enable this to see inline values in the editor while debugging."
```
Comment (luabud) [inline package.nls.json:8]:
```suggestion
    "debugpy.showPythonInlineValues": "Whether to display inline values in the editor while debugging."
```

**Comment by connor4312** [comment]
> Yes, it will disable all the inline values

Is that intended? That seems like a regression for folks who were using them up until this point

**Comment by karthiknadig** [comment]
@connor4312 This setting is similar to `debug.inlineValues` but it should not affect general one. The `debugpy.showPythonInlineValues` is added so we can explore the value filtering, via experimentation, till we get to a point where we can remove it. I agree that a more general purpose language specific version o `debug.inlineValue` is a core feature.

File: package.json [hunk 3]
```diff
@@ -131,6 +131,15 @@
                     "description": "%debugpy.debugJustMyCode%",
                     "scope": "resource",
                     "type": "boolean"
+                },
+                "debugpy.showPythonInlineValues": {
+                    "default": false,
+                    "description": "%debugpy.showPythonInlineValues%",
```
Comment (karthiknadig) [inline package.json:137]:
```suggestion
                    "description": "%debugpy.showPythonInlineValues.description%",
```

File: package.nls.json [hunk 4]
```diff
@@ -4,5 +4,6 @@
     "debugpy.command.debugUsingLaunchConfig.title": "Python Debugger: Debug using launch.json",
     "debugpy.command.reportIssue.title": "Report Issue...",
     "debugpy.command.viewOutput.title": "Show Output",
-    "debugpy.debugJustMyCode": "When debugging only step through user-written code. Disable this to allow stepping into library code."
+    "debugpy.debugJustMyCode": "When debugging only step through user-written code. Disable this to allow stepping into library code.",
+    "debugpy.showPythonInlineValues": "Whether to display inline values in the editor while debugging."
```
Comment (karthiknadig) [inline package.nls.json:8]:
```suggestion
    "debugpy.showPythonInlineValues.description": "Whether to display inline values in the editor while debugging."
```

**Review by karthiknadig — CHANGES_REQUESTED** [review]

**Comment by connor4312** [comment]
@karthiknadig I have not tested the changes in this PR, but in VS Code if there is an inline value provider registered for a language then that will be used instead of the default handling that VS Code does, and is doing for Python stable. If I'm understanding it right, the implementation in this PR of just returning an empty array from the provider won't fall back to the old behavior, but will instead turn off inline values for everyone, including those who might have been using them already in Python.

https://github.com/microsoft/vscode/blob/06643dd85264bc88ab30acc007baad0db33b8c71/src/vs/workbench/contrib/debug/browser/debugEditorContribution.ts#L676

I think the better implementation is not registering an inline value provider at all if the setting is turned off.

**Comment by karthiknadig** [comment]
Thanks @connor4312 for the explanation. I agree that this should skip registration, if the setting is set to disable. If the setting was changed when the extension is already activated, we can just use the on settings change event and dispose the provider. `registerInLineProvider` API returns a disposable that we can use to register/unregister dynamically.

File: src/extension/extensionInit.ts [hunk 5]
```diff
@@ -200,8 +202,32 @@ export async function registerDebugger(context: IExtensionContext): Promise<IExt
...
+        registerInlineValuesProviderDisposable = languages.registerInlineValuesProvider(
+            { language: 'python' },
+            new PythonInlineValueProvider(),
+        );
+        context.subscriptions.push(registerInlineValuesProviderDisposable);
+    }
+
+    context.subscriptions.push(
+        workspace.onDidChangeConfiguration((event: ConfigurationChangeEvent) => {
+            if (event.affectsConfiguration('debugpy')) {
```
Comment (connor4312) [inline src/extension/extensionInit.ts:218]: An aside but I have a [ConfigValue](https://github.com/microsoft/vscode-extension-test-runner/blob/9d6f9575f55f33fe2374d2d7af5f8eaacd81c7ff/src/configValue.ts#L21) helper that I've copied between multiple projects that makes this easier :)

**Review by connor4312 — COMMENTED** [review]
lgtm

File: src/extension/extensionInit.ts (hunk 5 above)
Comment (paulacamargo25) [reply src/extension/extensionInit.ts:218]: Thanks for the suggestion Connor, I'll review it and add it as a dev week item.

**Review by karthiknadig — APPROVED** [review]

**Pull request merged**